File: Data Structures & Algorithms/validate-parentheses/submission-0.py
class Solution:
    def isValid(self, s: str) -> bool:
        n=len(s)
        stack=[0]*n
        top=-1
        i=0
        valid=0
        while(i<n):
            if (s[i]=='[' or s[i]=='{' or s[i]=='('):
                top+=1
                stack[top]=s[i]
            elif(s[i]==']' or s[i]=='}' or s[i]==')'):
                ch =stack[top]
                top-=1
                if((s[i]==']' and ch=='[') or (s[i]=='}' and ch=='{') or (s[i]==')' and ch=='(')):
                    valid=1
                else:
                    break;
            i+=1
                    
        if valid==1 and top==-1:
            return True
        else:
            return False

File: Data Structures & Algorithms/minimum-stack/submission-0.java
class MinStack {
    int top=-1;
    int []stack=new int[10000];
    public MinStack() {

        
    }
    
    public void push(int val) {
        top++;
        stack[top]=val;
    }
    
    public void pop() {
        top--;
        
    }
    
    public int top() {
        return stack[top];
    }
    
    public int getMin() {
        int x=stack[0];
        for(int i=0;i<=top;i++){
            x=Math.min(stack[i],x);
        }return x;
        }
    }

